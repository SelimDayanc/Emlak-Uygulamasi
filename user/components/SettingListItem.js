/*
 @flow
 */
import React, {Component, PropTypes} from 'react';
import {StyleSheet, Text, TouchableHighlight, View} from 'react-native';
import colors from '../../common/colors';
import FontAwesome from 'react-native-vector-icons/FontAwesome';

export default class SettingListItem extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    route: PropTypes.string.isRequired,
    loadScene: PropTypes.func.isRequired,
    icon: PropTypes.string.isRequired,
  };

  render() {
    const {title, icon, route, loadScene} = this.props;
    return (
      <TouchableHighlight
        onPress={() => loadScene(route)}
        underlayColor="transparent"
        style={styles.container}>
        <View style={styles.rowContainer}>
          <View style={{flex: 9}}>
            <Text style={styles.title}>{title}</Text>
          </View>
          <View style={{flex: 1, alignItems: 'center'}}>
            <FontAwesome
              style={styles.icon}
              name={icon}
              color={colors.darkGrey}
              size={20}
            />
          </View>
        </View>
      </TouchableHighlight>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  rowContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: 20,
  },
  title: {
    color: colors.darkGrey,
    fontWeight: '200',
  },
  icon: {
    fontWeight: '100',
  },
});
