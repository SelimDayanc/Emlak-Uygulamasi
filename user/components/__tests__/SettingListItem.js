import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import SettingListItem from '../SettingListItem';

test('renders heart', () => {
  const tree = renderer
    .create(
      <SettingListItem
        title="title"
        route="test"
        loadScene={() => {}}
        icon="battery"
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
