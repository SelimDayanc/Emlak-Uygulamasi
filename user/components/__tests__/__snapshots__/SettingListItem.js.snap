// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders heart 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "flex": 1,
      },
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingVertical": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 9,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#484848",
            "fontWeight": "200",
          }
        }
      >
        title
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#484848",
              "fontSize": 20,
            },
            Object {
              "fontWeight": "100",
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        ï‰€
      </Text>
    </View>
  </View>
</View>
`;
